### PR TITLE
api/vmalertmanager: adds webserver configuration

### PR DESCRIPTION
**Pull request by f41gh7**
Adds a new type for generic webserver TLS configuration.

* First of all, it allows to properly configure alertmanager and detect tls related params for it. It fixes scheme detection for probes, AsURL function.

* Second, it allows later to rollout this type to other application and have a generic way of TLS/mTLS configuration

* Fixes bug with tlsAssets mount with custom config reloader for alertmanager. Previously it was not mounted correctly and may cause issues with inaccessible assets.

related issues:
- https://github.com/VictoriaMetrics/operator/issues/994
- https://github.com/VictoriaMetrics/operator/issues/1033

### CODE REVIEW

File: api/operator/v1beta1/vmalertmanager_types.go [hunk 1]
```diff
@@ -484,8 +490,12 @@ func (cr *VMAlertmanager) ProbePort() string {
 	return cr.Spec.PortName
 }
 
+// ProbeScheme returns scheme for probe
 func (cr *VMAlertmanager) ProbeScheme() string {
-	return strings.ToUpper(protoFromFlags(cr.Spec.ExtraArgs))
+	if cr.Spec.WebConfig != nil && cr.Spec.WebConfig.TLSServerConfig != nil {
```
Comment (AndrewChubatiuk) [inline api/operator/v1beta1/vmalertmanager_types.go:495]: not sure that this validation is enough. user can set `WebserverTLSConfig.ClientAuthType` to `NoClientCert`

Comment (AndrewChubatiuk) [reply api/operator/v1beta1/vmalertmanager_types.go:495]: is there any validation on alertmanager site?

Comment (f41gh7) [reply api/operator/v1beta1/vmalertmanager_types.go:495]: In this case, tls must be configured. 

 This PR lacks of:
- crd server-side validation
- validation webhook checks
- mention of changes at changelog

It should be addressed before merging.

Comment (f41gh7) [reply api/operator/v1beta1/vmalertmanager_types.go:495]: updated checks, added validation and changelog entry

**Review by AndrewChubatiuk — APPROVED** [review]

**Pull request merged**